Automattic/woocommerce-payments
# Add support phone and email in the admin settings page

Fixes #3981 

#### Changes proposed in this Pull Request

- Extract two new components `SupportEmailInput` and `SupportPhoneInput` from `client/card-readers/settings/sections/contacts-details.js`. 
- With that, we can use them for both `Card Readers` setting page and WCPay general settings pages.
- Add tests for the new components.









| Before      | After |
| ----------- | ----------- |
| ![settings-before](https://user-images.githubusercontent.com/10045087/215710469-1248b4e6-a63d-4811-b6a4-f7883432a520.png)  | ![settings-after-with-arrow](https://user-images.githubusercontent.com/10045087/215711742-72bc4215-4399-4e69-ac25-24a114a5004f.png)  |

#### Testing instructions

- Checkout this PR, and run `npm start`.
- Visit wp-admin > Payments > Settings.
- See two new input fields.
- Edit and update them.

If possible, try to test with `Card Readers` settings page:

- In your local server, return `true` for `has_card_readers_available` flag/method, like this 2ef2d-pb
- Visit wp-admin > Payments > `Card Readers` > Receipt Details.
- Edit and update the same fields before and after this PR.





#### Note 1 

GitHub action throws a SOS warning about increasing sizes of JS and CSS for the admin settings page. I think this is fine as it brings a nice UI and validation for the phone input. Under the hood, it uses the `PhoneNumberInput` component, which is [based on 'intl-tel-input' package](https://github.com/Automattic/woocommerce-payments/blob/ef6339ff5a79ad8e26bf6f9c4f2ab8445b913310/client/settings/phone-input/index.tsx#L6-L6).

#### Note 2

> Full Stripe accounts allow the phone to be toggled off, so it would be great if WooCommerce Payments can offer that also.
> - From this comment https://github.com/Automattic/woocommerce-payments/issues/3981#issuecomment-1072658960

With the connected account, there is no option for this. Stripe always requires this number and here is the most detail I can get when trying to update this number via `curl`, Stripe Dashboard, and via API call through local server: 

![Markup on 2023-01-31 at 16:00:16](https://user-images.githubusercontent.com/10045087/215715309-87fca2a0-4c4e-42eb-a267-baba836f61a7.png)

-------------------


**Post merge**
